STM32WB0: add timers support

mathieuchopstm opened the pull request:
Add support for the timer peripherals of STM32WB0 series (both `counter` and `pwm`).

Tests passed successfully on Nucleo-WB05KZ, Nucleo-WB07CC and Nucleo-WB09KE:
* PWM
  * `samples/basic/fade_led/`
  * `samples/basic/blinky_pwm/`
  * `tests/drivers/pwm/pwm_api/` (`drivers.pwm`)
* Counter (excl. WB07)
  * `tests/drivers/counter/counter_basic_api/` (`drivers.counter.basic_api`)

mathieuchopstm committed:
dts: arm: st: wb0: add timers

Add nodes for all timer peripherals to DTSI of the STM32WB0 series.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
drivers: pwm: stm32: add support for STM32WB0

This commit adds support for the STM32WB0 series to
the existing STM32 PWM driver.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
drivers: counter: stm32: add support for STM32WB0

This commit adds support for the STM32WB0 series to
the existing STM32 timer-based counters driver.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
boards: st: nucleo_wb09ke: add PWM support and LED

Add the PWM LED to Nucleo-WB09KE board and mark as compatible with PWM.
Also mark counters as supported (but don't enable by default).

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
boards: st: nucleo_wb05kz: add PWM support and LED

Add the PWM LED to Nucleo-WB05KZ board and mark as compatible with PWM.
Also mark counters as supported (but don't enable by default).

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm commented:
Blocking merge until #83524 is integrated on main to avoid needing a follow-up PR for WB07 only.

erwango commented:
Please update  following https://github.com/zephyrproject-rtos/zephyr/pull/83524 merge

mathieuchopstm committed:
boards: st: nucleo_wb07cc: add PWM support and LED

Add the PWM LED to Nucleo-WB07CC board and mark as compatible with PWM.
Also mark counters as supported (but don't enable by default).

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm commented:
Added missing `counter` nodes for TIM1/TIM2, and support for PWM on Nucleo-WB07.
